tests/Testcontainers.Tests/Unit/Configurations/WaitUntilContainerIsHealthyTest.cs

```diff
@@ -0,0 +1,45 @@
+namespace DotNet.Testcontainers.Tests.Unit.Configurations
+{
+  using System;
+  using System.IO;
+  using System.Threading.Tasks;
+  using DotNet.Testcontainers.Builders;
+  using DotNet.Testcontainers.Configurations.WaitStrategies;
+  using DotNet.Testcontainers.Containers;
+  using Xunit;
+
+  public sealed class WaitUntilContainerIsHealthyTest
+  {
+    [Fact]
+    public async Task StartsOnceHealthy()
+    {
+      var imageName = await new ImageFromDockerfileBuilder()
```

Comment:
Can we pass the image as class fixture? We can use the same image to test *fail* and *ok*, by overriding the entrypoint or set an env variable to wait until / write the file?